xroad-catalog-collector/src/main/java/org/niis/xroad/catalog/collector/tasks/ListClientsTask.java

```diff
@@ -0,0 +1,127 @@
+import org.niis.xroad.catalog.collector.wsimport.ClientType;
+import org.niis.xroad.catalog.collector.wsimport.XRoadObjectType;
+import org.niis.xroad.catalog.persistence.entity.ErrorLog;
+import org.niis.xroad.catalog.persistence.entity.Member;
+import org.niis.xroad.catalog.persistence.entity.MemberId;
+import org.niis.xroad.catalog.persistence.entity.Subsystem;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+
+@Slf4j
+@Component
+public class ListClientsTask implements Runnable {
+
+    private final TaskPoolConfiguration taskPoolConfiguration;
+    private final CatalogService catalogService;
+    private final Queue<ClientType> listMethodsQueue;
+    private final NewMembersEventPublisher newMembersEventPublisher;
+
+    public ListClientsTask(CatalogService catalogService, TaskPoolConfiguration taskPoolConfiguration, Queue<ClientType> listMethodsQueue,
+                           NewMembersEventPublisher newMembersEventPublisher) {
+        this.taskPoolConfiguration = taskPoolConfiguration;
+        this.catalogService = catalogService;
+        this.listMethodsQueue = listMethodsQueue;
+        this.newMembersEventPublisher = newMembersEventPublisher;
+    }
+
+    public void run() {
+        log.info("Starting ListClientsTask");
+        if (CollectorUtils.isTimeBetweenHours(taskPoolConfiguration.getFlushLogTimeAfterHour(),
+                taskPoolConfiguration.getFlushLogTimeBeforeHour())) {
+            catalogService.deleteOldErrorLogEntries(taskPoolConfiguration.getErrorLogLengthInDays());
+        }
+
+        if (taskPoolConfiguration.isFetchRunUnlimited()
+                || CollectorUtils.isTimeBetweenHours(taskPoolConfiguration.getFetchTimeAfterHour(),
+                taskPoolConfiguration.getFetchTimeBeforeHour())) {
+            fetchClients();
+        }
+    }
+
+    private void fetchClients() {
+        String listClientsUrl = taskPoolConfiguration.getListClientsHost() + "/listClients";
+        try {
+            log.info("Getting client list from {}", listClientsUrl);
+            ClientList clientList = ClientListUtil.clientListFromResponse(listClientsUrl);
+            HashMap<MemberId, Member> m = populateMapWithMembers(clientList);
+            Set<Member> newMembers = catalogService.saveAllMembersAndSubsystems(m.values());
+
+            // We only fetch WSDL-s and REST services from subsystems
+            List<ClientType> subsystems = clientList.getMember().stream()
+                    .filter(client -> XRoadObjectType.SUBSYSTEM.equals(client.getId().getObjectType()))
+                    .toList();
+            listMethodsQueue.addAll(subsystems);
+
+            log.info("All subsystems ({}) sent to ListMethodsTask", subsystems.size());
+
+            newMembersEventPublisher.publishNewMembersEvent(newMembers.stream().map(Member::getMemberCode).toList());
```

Comment:
The previous implementation also used `toList()` here, but now looking at it I think it would make more sense to use a `Set` instead, since we don't want duplicates.